workspaces/ui-v2/src/pages/docs/components/ShapeEditor.tsx

```diff
@@ -0,0 +1,465 @@
+import React, { FC, useMemo } from 'react';
+import {
+  makeStyles,
+  darken,
+  lighten,
+  IconButton,
+  ButtonGroup,
+  Button,
+  TextField,
+} from '@material-ui/core';
+import { Check as CheckIcon } from '@material-ui/icons';
+import ClassNames from 'classnames';
+import Color from 'color';
+import DeleteOutlineIcon from '@material-ui/icons/DeleteOutline';
+
+import { IFieldDetails, IShapeRenderer } from '<src>/types';
+import { JsonType } from '@useoptic/optic-domain';
+import * as Theme from '<src>/styles/theme';
+
+export const ShapeEditor: FC<{
+  fields: IFieldDetails[];
+  selectedFieldId: string | null;
+  setSelectedField: (fieldId: string | null) => void;
+}> = ({ fields, selectedFieldId, setSelectedField }) => {
+  const classes = useStyles();
+
+  let onChangeFieldDescription = useMemo(
+    () => (fieldId: string, description: string) => {
+      console.log('description changed for field', fieldId, description);
+    },
+    []
+  );
+
+  return (
+    <div className={classes.container}>
+      {fields.length > 0 && (
+        <ul className={classes.rowsList}>
+          {fields.map((field) => (
+            <li className={classes.rowListItem}>
+              <Row
+                field={field}
+                description={''} // TODO FLEB: wire me in
+                selected={selectedFieldId === field.fieldId}
+                onChangeDescription={onChangeFieldDescription}
+                onSelect={setSelectedField}
+              />
+            </li>
+          ))}
+        </ul>
+      )}
+    </div>
+  );
+};
+
+const Row: FC<{
+  field: IFieldDetails;
+  description: string;
+  selected: boolean;
+
+  onChangeDescription?: (fieldId: string, description: string) => void;
+  onSelect?: (fieldId: string) => void;
+}> = function ShapeEditorRow({
+  field,
+  description,
+  selected,
+  onChangeDescription,
+  onSelect,
+}) {
+  const classes = useStyles();
+
+  const onClickFieldHeader = useMemo(
+    () => () => {
+      if (onSelect) onSelect(field.fieldId);
```

Comment:
We'll want to unselect here - so `onSelect(selected ? null : field.fieldId)` which will unhighlight when dismissed (such that the highlight action is undoable)

Reply:
Sure, that's probably fine. Just make sure to stop propagation on the remove control, so we don't close inadvertently 😅.